src/app/Fake.DotNet.MSBuild/MSBuild.fs

```diff
@@ -872,24 +867,29 @@ module MSBuild =
           yield! properties ]
         |> Seq.choose id
         |> Seq.map (fun (k, v) -> "/" + k + (if String.isNullOrEmpty v then "" else ":" + v))
-        |> Args.toWindowsCommandLine
+        |> Seq.toList
 
     /// [omit]
     let buildArgs (setParams: MSBuildParams -> MSBuildParams) =
+        let p = MSBuildParams.Create() |> setParams
+        p, fromCliArguments p.CliArguments |> Args.toWindowsCommandLine
+
+    /// [omit]
```

Comment:
no need to `omit` comment, if the function will not make it to documentation then u can use visibility modified, maybe change it to internal? Also, can you come up with another name instead of `buildArgs2`. While reading this, I had to search if `buildArgs` exists and compare both of them to see the difference.